walletnode/api/design/artworks.go

```diff
@@ -307,3 +340,190 @@ var RegisterTaskPayload = Type("RegisterTaskPayload", func() {
 	})
 	Required("taskId")
 })
+
+// SearchArtworkParams are query params to searchArtwork request
+var SearchArtworkParams = func() {
+	Attribute("artist", String, func() {
+		Description("Artist PastelID or special value; mine")
+		MaxLength(256)
+	})
+	Attribute("limit", Int, func() {
+		Description("Number of results to be return")
+		Minimum(10)
+		Default(10)
+		Example(10)
+	})
+
+	Attribute("artist_name", String, func() {
+		Description("search for matches in ticket's field - artist_name")
+		MaxLength(256)
+	})
+	Attribute("term", String, func() {
+		Description("search for matches in ticket's field - artwork_title")
+		MaxLength(256)
+	})
+	Attribute("series", String, func() {
+		Description("search for matches in ticket's field - artwork_series_name")
+		MaxLength(256)
+	})
+	Attribute("descr", String, func() {
+		Description("search for matches in ticket's field - artist_written_statement")
+		MaxLength(256)
+	})
+
+	Attribute("keyword", String, func() {
+		Description("search for matches in ticket's field - artwork_keyword_set")
+		MaxLength(256)
+	})
+
+	Attribute("min_block", Int, func() {
+		Description("return art with art min_block <= activation ticket txid ")
+	})
+
+	Attribute("max_block", Int, func() {
+		Description("return art with art activation ticket txid <= max_block")
+	})
+
+	Attribute("min_copies", Int, func() {
+		Description("return art with min_copies <= number of created copies")
+	})
+
+	Attribute("max_copies", Int, func() {
+		Description("return art with number of created copies <= max_copies")
+	})
+
+	Attribute("min_nsfw_score", Int, func() {
+		Description("return art with min_nsfw_score<=nsfw_score")
+	})
+
+	Attribute("max_nsfw_score", Int, func() {
+		Description("return art with nsfw_score <= max_nsfw_score")
+	})
+
+	Attribute("min_rareness_score", Int, func() {
+		Description("return art with min_rareness_score <= rareness_score ")
+	})
+
+	Attribute("max_rareness_score", Int, func() {
+		Description("return art wit rareness_score <= max_rareness_score")
+	})
+}
+
+// ArtworkSearchResult  is response of artwork search.
+var ArtworkSearchResult = Type("ArtworkSearchResult", func() {
```

Comment:
Unfortunately you didn't understand what need to be returned, re-read @ayk [message]( https://blockchain-devel.slack.com/archives/C022CA67U69/p1622173995106800) and ask him if it's not clear for you